arrows/vtk/applets/color_mesh.cxx

```diff
@@ -0,0 +1,579 @@
+    set_output(mesh);
+    set_frame(frame_);
+    set_frame_sampling(frame_sampling_);
+    set_all_frames(all_frames_);
+    colorize();
+    LOG_INFO(main_logger, "Save mesh file...");
+    if (! save_mesh(mesh, output_mesh_.c_str(), ply_color_.c_str()))
+    {
+      return false;
+    }
+    return true;
+  }
+
+  void report_progress_changed(
+    std::string const& message, int percentage) override
+  {
+        LOG_INFO(main_logger,
+                 message << ": " << percentage << "%");
+  }
+};
+
+// ----------------------------------------------------------------------------
+int
+color_mesh::
+run()
+{
+
+  try
+  {
+    switch (d->process_command_line(command_args()))
+    {
+    case priv::HELP:
+      std::cout << m_cmd_options->help();
+      return EXIT_SUCCESS;
+    case priv::WRITE:
+      return EXIT_SUCCESS;
+    case priv::FAIL:
+      return EXIT_FAILURE;
+    case priv::SUCCESS:
+      ;
+    }
+
+    if (d->config_ == nullptr)
+    {
+      return EXIT_FAILURE;
+    }
+
+    d->initialize();
+    if (!d->run_algorithm())
+    {
+      return EXIT_FAILURE;
+    }
+
+    return EXIT_SUCCESS;
+  }
+  catch (std::exception const& e)
+  {
+    LOG_ERROR(main_logger, "Exception caught: " << e.what());
+
+    return EXIT_FAILURE;
+  }
+  catch (...)
+  {
+    LOG_ERROR(main_logger, "Unknown exception caught");
+
+    return EXIT_FAILURE;
+  }
+} // run
+
+// ----------------------------------------------------------------------------
+void
+color_mesh::
+add_command_options()
+{
+  m_cmd_options->custom_help( wrap_text(
+    "[options] input-mesh video-file cameras-dir output-mesh\n"
+    "This program colors an input-mesh from a video (or list of images) and "
+    "a list of camera files stored in a directory. A mesh colored with "
+    "the average color or with a color for a particular camera is produced.") );
+
+  m_cmd_options->positional_help("\n  input-mesh  - input mesh file."
+                                 "\n  video-file  - input video file."
+                                 "\n  cameras-dir  - input camera directory."
+                                 "\n  output-mesh - output mesh file."
+  );
+
+  m_cmd_options->add_options()
+    ( "a,all-frames",
+      "Compute average color or save each frame color",
+      cxxopts::value<bool>()->default_value("false") )
+    ( "c,config",
+      "Configuration file for tool",
+      cxxopts::value<std::string>() )
+    ( "f,frame",
+      "Frame index to use for coloring. "
+      "If -1 use an average color for all frames.",
+      cxxopts::value<int>()->default_value( "-1"))
+    ( "h,help",     "Display applet usage" )
+    ( "m,mask-file",
+      "An input mask video or list of mask images to indicate "
+      "which pixels to ignore.",
+      cxxopts::value<std::string>())
+    ( "o,output-config",
+      "Output a configuration. This may be seeded with a "
+      "configuration file from -c/--config.",
+      cxxopts::value<std::string>() )
+    ( "p,ply-color",
```

Comment:
As discussed, I'd recommend calling this "active-color-mode" or something similar.  I'm not sure what the abbreviated flag should be since we've already used "a", "c", and "m".